Remove double to string convertion in FormData

doochik opened the pull request:
These convertions introduced by 2 different PRs
https://github.com/facebook/react-native/commit/5061fde3173e6228d544b5df1ce8ce4e2a7dd2a4
https://github.com/facebook/react-native/commit/de392b5351494675220e0ef5176d8a8f3d473477

olegbl commented:
Make sure it works for booleans, etc... Not every non-string is an object. (And { ...value } crashes when value is a boolean.)

facebook-github-bot-6 commented:
@olegbl would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

satya164 commented on Libraries/Network/FormData.js:
```diff
@@ -85,7 +81,7 @@ class FormData {
       }
-      // Cast to string all other values
+      // Convert non-object values to strings as per FormData.append() spec
       return {string: String(value), headers, fieldName: name};
```
We should probably remove this extra conversion instead of removing the one at the top

doochik replied:
why? there are identically

satya164 replied:
Check @olegbl 's comment

doochik commented:
@olegbl @satya164 
Do you mean cases like `formData.append('foo', new Boolean(true))`?
No, there is no crashes. Just value is empty in `formData.getParts()` cause no iteratable props in `new Boolean()`.

I can rewrite this https://github.com/facebook/react-native/blob/bef175af61d387084fd9161e495cac3e1b414556/Libraries/Network/FormData.js#L79 to something like `if (isPlainObject(value)) { ... }`. But I think it's overhead.

In this PR i've just removed some duplicate code
* value casting at line 69: `value = '' + value` and line 89: `value: String(value)`
* checking value type at line 68: `if (typeof value !== 'object')` and line 79: `if (typeof value === 'object')`).

I've removed first conversion (`value = '' + value`) cause `String(value)` is more obviously in my opinion.

olegbl commented:
I meant formData.append('foo', true). Looks like it should be fine since "if (typeof value === 'object') {" got added some time back.

doochik commented:
Yep, it should be fine and will be casted to `foo=true` in HTTP request

satya164 commented:
Can you rebase against master so that the tests pass?

doochik commented:
i've rebased my branch

facebook-github-bot-3 commented:
@doochik do you have any updates for this pull request? It's been a while so we wanted to check in and see if you've looked at the requested changes.

satya164 commented:
@facebook-github-bot shipit
